src/http/server/handlers/static_file_handler.cr

```diff
@@ -64,6 +64,23 @@ class HTTP::StaticFileHandler
       context.response.content_type = "text/html"
       directory_listing(context.response, request_path, file_path)
     elsif is_file
+      last_modified = File.stat(file_path).mtime
+      context.response.headers["Last-Modified"] = HTTP.rfc1123_date(last_modified)
+
+      if if_modified_since = context.request.headers.fetch("If-Modified-Since", nil)
+        # TODO: Use a more generalized time format parser for better compatibility to RFC 7232
+        header_time = Time.parse(if_modified_since, "%a, %d %b %Y %H:%M:%S GMT")
+
+        # File mtime probably has a higher resolution than the header value.
+        # An exact comparison might be slightly off, so we add 1s padding.
+        # Static files should generally not be modified in subsecond intervals, so this is perfectly safe.
+        # This might replaced by a more sophisticated time comparison when it becomes available.
+        if last_modified <= header_time + 1.second
```

Comment:
Maybe we can use `last_modified.total_seconds.to_i == header_time.total_seconds.to_i`?

Reply:
I'm not sure if this could lead to failing edge cases because of rounding errors.

Reply:
This would certainly not have the expected results because a time at `0.999_999_999 s` will be just one nanosecond away from `1.0s` but `total_seconds != total_seconds`.

Reply:
While it's not super critical here, it doesn't make sense to sacrifice precision for performance since the whole point is to improve efficiency if the time stamp is pretty recent.